qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -913,17 +914,23 @@ def _run_analysis(
                     **fit_opt.options,
                 )
                 fit_results.append(fit_result)
-
-            # Find best value with chi-squared value
-            if len(fit_results) == 0:
-                raise AnalysisError(
-                    "All initial guesses and parameter boundaries failed to fit the data. "
-                    "Please provide better initial guesses or fit parameter boundaries."
-                )
-            fit_result = sorted(fit_results, key=lambda r: r.reduced_chisq)[0]
-
-        except AnalysisError:
+            except AnalysisError:
+                # some guess might be completely off from true parameters.
+                # even if some attempts fail, we just ignore that result and continue
+                # as long as we have more fit option candidates.
```

Comment:
```suggestion
                # Some guesses might be too far from the true parameters and may thus fail.
                # We ignore initial guesses that fail and continue with the next fit candidate.
```
I removed the `as long as we have more fit options` as the following warning is explicit enough.